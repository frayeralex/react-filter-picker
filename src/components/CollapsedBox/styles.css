.CollapsedBox {
  display: block;
}

.CollapsedBox_expanded .CollapsedBox_content {
  height: auto;
}

.CollapsedBox_header {
  padding: 10px;
  background: rgba(0,0,0,.1);
  display: flex;
  border-top: 1px solid #000;
}

.CollapsedBox_content {
  display: flex;
  flex-wrap: wrap;
  overflow: hidden;
  height: 0;
}

.CollapsedBox_title {
  flex-grow: 1;
}

.CollapsedBox_indicator {
  font-size: 20px;
  font-weight: 600;
}
