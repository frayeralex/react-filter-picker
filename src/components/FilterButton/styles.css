.FilterButton {
  box-sizing: content-box;
  border: 1px solid #000;
  padding: 10px;
  margin: 0 5px;
  min-width: 60px;
  text-align: center;
  flex-grow: 1;
}

.FilterButton_active {
  background: rgba(0, 0,0, .1);
}
