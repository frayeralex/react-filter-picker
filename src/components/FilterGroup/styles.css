.FilterGroup {
  padding: 10px;
}

.FilterGroup_items_container {
  display: flex;
  flex-wrap: wrap;
}

.FilterGroup_controls_container {
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.FilterGroup_controls_container_btn {
  background: #fff;
  padding: 5px 20px;
  border-radius: 0;
}
