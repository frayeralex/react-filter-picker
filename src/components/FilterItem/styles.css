.FilterItem {
  border: 1px solid #000;
  padding: 5px;
  margin: 2px;
  cursor: pointer;
}

.FilterItem_active {
  background: rgba(0, 0,0, .1);
}
