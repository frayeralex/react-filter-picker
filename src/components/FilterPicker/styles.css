.FilterPiker {
  width: 300px;
  position: absolute;
  top: 0;
  left: 0;
  background: #f1f1f1;
  transition: .4s;
  animation: fadeIn .3s;
  transform-origin: bottom center;

}

.FilterPiker_triangular {
  position: absolute;
  width: 0;
  height: 0;
  border-left: 10px solid transparent;
  border-right: 10px solid transparent;
  border-bottom: 10px solid rgba(0, 0, 0, .1);
  animation: fadeIn .3s;
  transform-origin: bottom center;
  transition: .2s;
}

.FilterPiker_content {
  display: flex;
  flex-direction: column;
}

@keyframes fadeIn {
  from {
    opacity: 0;
    transform: translateY(20px);
  }
  to {
    transform: translateY(0)
  }
}
