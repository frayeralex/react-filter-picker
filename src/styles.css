/* add css styles here (optional) */

.Filter {
  width: 100%;
}

.Filter_buttons_container {
  display: flex;
  align-items: center;
}

.FilterItem {
  border: 1px solid #000;
  padding: 10px;
  margin: 10px;
}
