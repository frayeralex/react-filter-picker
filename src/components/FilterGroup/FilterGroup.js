import React, { PureComponent } from 'react'
import { FilterItem } from '../index'
import styles from './styles.css';

class FilterGroup extends PureComponent {
  static defaultProps = {
    selectedFilters: [],
  }
  constructor(props) {
    super(props);

    this.state = {
      selectedFilters: props.selectedFilters
    }
  }

  handleFilterItemClick = (id) => () => {
    this.setState({
      selectedFilters: this.state.selectedFilters.includes(id)
        ? this.state.selectedFilters.filter(item => item !== id)
        : [...this.state.selectedFilters, id]
    });
  }

  handleApplyClick = () => {
    this.props.onApply(this.props.filter, this.state.selectedFilters);
  }

  render() {
    return (
      <div className={styles.FilterGroup}>
        <div className={styles.FilterGroup_items_container}>
          {this.props.filterItems.map(item => {
            return (
              <FilterItem
                key={item.id}
                title={item.title}
                active={this.state.selectedFilters.includes(item.id)}
                onClick={this.handleFilterItemClick(item.id)}
              />
            )
          })}
        </div>
        <div className={styles.FilterGroup_controls_container}>
          <button
            className={styles.FilterGroup_controls_container_btn}
            type="button"
            onClick={this.handleApplyClick}>
            Apply
          </button>
        </div>
      </div>
    )
  }
}

export default FilterGroup;
