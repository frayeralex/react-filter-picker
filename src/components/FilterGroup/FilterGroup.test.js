import React from 'react';
import FilterGroup from './FilterGroup';
import { FilterItem } from '../index';
import styles from './styles.css';

const props = {
  filter: 'colors',
  filterItems: [
    {
      id: '1',
      title: 'First',
    },
    {
      id: '2',
      title: 'Second',
    },
  ],
};

describe('<FilterGroup/>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<FilterGroup {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should show FilterItems correctly', () => {
    const wrapper = shallow(<FilterGroup {...props} />);
    wrapper.setProps({
      filterItems: [...props.filterItems, { id: '3', title: '3' }],
    });
    expect(wrapper.find(FilterItem)).toHaveLength(3);
    expect(wrapper).toMatchSnapshot();
  });
  it('should set default selectedFilters state correctly', () => {
    const selectedFilters = ['1', '2'];
    const wrapper = shallow(
      <FilterGroup {...props} selectedFilters={selectedFilters} />,
    );
    expect(wrapper.state()).toEqual({ selectedFilters });
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle apply click correctly', () => {
    const onApply = jest.fn();
    const selectedFilters = ['1'];
    const wrapper = shallow(<FilterGroup {...props} onApply={onApply} />);
    wrapper
      .find('button')
      .first()
      .simulate('click', {});
    expect(onApply).toHaveBeenCalledWith(props.filter, []);
    wrapper.setState({ selectedFilters });
    expect(wrapper.state()).toEqual({ selectedFilters });
    wrapper
      .find('button')
      .first()
      .simulate('click', {});
    expect(onApply).toHaveBeenLastCalledWith(props.filter, ['1']);
  });
  it('should handle FilterItem click correctly', function() {
    const wrapper = shallow(<FilterGroup {...props} />);
    wrapper
      .find(FilterItem)
      .first()
      .shallow()
      .simulate('click');
    expect(wrapper.state()).toEqual({ selectedFilters: ['1'] });
    wrapper
      .find(FilterItem)
      .at(1)
      .shallow()
      .simulate('click');
    expect(wrapper.state()).toEqual({ selectedFilters: ['1', '2'] });
    wrapper
      .find(FilterItem)
      .at(0)
      .shallow()
      .simulate('click');
    expect(wrapper.state()).toEqual({ selectedFilters: ['2'] });
    wrapper
      .find(FilterItem)
      .at(1)
      .shallow()
      .simulate('click');
    expect(wrapper.state()).toEqual({ selectedFilters: [] });
  });
});
