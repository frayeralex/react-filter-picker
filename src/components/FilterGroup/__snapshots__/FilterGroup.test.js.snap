// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterGroup/> should render without crashing 1`] = `
<div>
  <div>
    <FilterItem
      active={false}
      key="1"
      onClick={[Function]}
      title="First"
    />
    <FilterItem
      active={false}
      key="2"
      onClick={[Function]}
      title="Second"
    />
  </div>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Apply
    </button>
  </div>
</div>
`;

exports[`<FilterGroup/> should set default selectedFilters state correctly 1`] = `
<div>
  <div>
    <FilterItem
      active={true}
      key="1"
      onClick={[Function]}
      title="First"
    />
    <FilterItem
      active={true}
      key="2"
      onClick={[Function]}
      title="Second"
    />
  </div>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Apply
    </button>
  </div>
</div>
`;

exports[`<FilterGroup/> should show FilterItems correctly 1`] = `
<div>
  <div>
    <FilterItem
      active={false}
      key="1"
      onClick={[Function]}
      title="First"
    />
    <FilterItem
      active={false}
      key="2"
      onClick={[Function]}
      title="Second"
    />
    <FilterItem
      active={false}
      key="3"
      onClick={[Function]}
      title="3"
    />
  </div>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Apply
    </button>
  </div>
</div>
`;
